### PR TITLE
Bot API 7.5

### PR DESCRIPTION
**Pull request by pxp9**
https://core.telegram.org/bots/api#june-18-2024

### CODE REVIEW

File: src/objects.rs [hunk 1]
```diff
@@ -4096,6 +4096,69 @@ pub struct InaccessibleMessage {
...
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Builder)]
+pub struct TransactionPartnerOther {
+    pub r#type: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Builder)]
+pub struct StarTransaction {
+    pub id: String,
+    pub amount: i32,
+    // u64 since it is Unix Timestamp and
```
Comment (ayrat555) [inline src/objects.rs:4150]: this comment can be removed. we're already using u64 for all dates

File: src/objects.rs [hunk 2]
```diff
@@ -4096,6 +4096,69 @@ pub struct InaccessibleMessage {
...
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Builder)]
+pub struct TransactionPartnerOther {
+    pub r#type: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Builder)]
+pub struct StarTransaction {
+    pub id: String,
```
Comment (ayrat555) [inline src/objects.rs:4148]: please add     #[builder(setter(into))] before this field.

please add it for other string fields as well

File: src/objects.rs [hunk 3]
```diff
@@ -4096,6 +4096,69 @@ pub struct InaccessibleMessage {
...
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Builder)]
+pub struct TransactionPartnerUser {
+    pub r#type: String,
+    pub user: User,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Builder)]
+pub struct TransactionPartnerOther {
+    pub r#type: String,
```
Comment (ayrat555) [inline src/objects.rs:4143]: please convert `TransactionPartner`  to enum

**Review by ayrat555 — CHANGES_REQUESTED** [review]

File: src/objects.rs (hunk 2 above)
Comment (pxp9) [reply src/objects.rs:4156]: [address comments](https://github.com/ayrat555/frankenstein/pull/174/commits/3e9b389ce4eb183ed04ab22dc89b2ce59e296ad0)

File: src/objects.rs (hunk 3 above)
Comment (pxp9) [reply src/objects.rs:4150]: [address comments](https://github.com/ayrat555/frankenstein/pull/174/commits/3e9b389ce4eb183ed04ab22dc89b2ce59e296ad0)

File: src/objects.rs (hunk 1 above)
Comment (pxp9) [reply src/objects.rs]: [address comments](https://github.com/ayrat555/frankenstein/pull/174/commits/3e9b389ce4eb183ed04ab22dc89b2ce59e296ad0)

File: src/objects.rs [hunk 4]
```diff
@@ -4096,6 +4096,75 @@ pub struct InaccessibleMessage {
...
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Builder)]
+pub struct RevenueWithdrawalStateSucceeded {
+    #[builder(setter(into))]
+    pub r#type: String,
+    pub date: u64,
+    url: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
```
Comment (ayrat555) [inline src/objects.rs:4126]: please remove type fields from these structs and add `#[serde(tag = "type", rename_all = "snake_case")]` to the enum

you can check how other enums are implemented

Comment (pxp9) [reply src/objects.rs:4121]: [tag with serde](https://github.com/ayrat555/frankenstein/pull/174/commits/13e7b69887d6888880f43564d98cf20259afdc62)

**Pull request merged**